src/cubit_utils.py

```diff
@@ -0,0 +1,294 @@
+import cubit
+from pathlib import Path
+import os
+import inspect
+import subprocess
+
+
+def init_cubit(cubit_flag):
+    """Initializes Coreform Cubit with the DAGMC plugin.
+
+    Arguments:
+        cubit_flag (bool): flag to indicate whether Coreform Cubit has been
+            initialized.
+
+    Returns:
+        cubit_flag (bool): flag to indicate whether Coreform Cubit has been
+            initialized.
+    """
+    if not cubit_flag:
+        cubit_plugin_dir = (
+            Path(os.path.dirname(inspect.getfile(cubit))) / Path('plugins')
+        )
+        cubit.init([
+            'cubit',
+            '-nojournal',
+            '-nographics',
+            '-information', 'off',
+            '-warning', 'off',
+            '-commandplugindir',
+            str(cubit_plugin_dir)
+        ])
+        cubit_flag = True
+
+    return cubit_flag
+
+
+def import_step_cubit(filename, import_dir):
+    """Imports STEP file into Coreform Cubit.
+
+    Arguments:
+        filename (str): name of STEP input file, excluding '.step' extension.
+        import_dir (str): directory from which to import STEP file.
+
+    Returns:
+        vol_id (int): Cubit volume ID of imported CAD solid.
+    """
+    import_path = Path(import_dir) / Path(filename).with_suffix('.step')
+    cubit.cmd(f'import step "{import_path}" heal')
+    vol_id = cubit.get_last_id("volume")
+
+    return vol_id
+
+
+def export_step_cubit(filename, export_dir=''):
+    """Export CAD solid as a STEP file via Coreform Cubit.
+
+    Arguments:
+        filename (str): name of STEP output file, excluding '.step' extension.
+        export_dir (str): directory to which to export the STEP output file
+            (defaults to empty string).
+    """
+    export_path = Path(export_dir) / Path(filename).with_suffix('.step')
+    cubit.cmd(f'export step "{export_path}" overwrite')
+
+
+def export_tet_mesh_cubit(volume_ids, filename, export_dir=''):
+    """Creates tetrahedral mesh of CAD volumes, exporting to Exodus format via
+    Coreform Cubit and converting to H5M via MOAB.
+
+    Arguments:
+        volume_ids (iterable of int): iterable of Cubit volume IDs.
+        filename (str): name of H5M output file, excluding '.h5m' extension.
+        export_dir (str): directory to which to export the H5M output file
+            (defaults to empty string).
+    """
+    for vol in volume_ids:
+        cubit.cmd(f'volume {vol} scheme tetmesh')
+        cubit.cmd(f'mesh volume {vol}')
+
+    exo_path = Path(export_dir) / Path(filename).with_suffix('.exo')
+    h5m_path = Path(export_dir) / Path(filename).with_suffix('.h5m')
+
+    cubit.cmd(f'export mesh "{exo_path}" overwrite')
+    subprocess.run(f'mbconvert {exo_path} {h5m_path}', shell=True)
+    Path.unlink(exo_path)
+
+
+def orient_spline_surfaces(volume_id):
```

Comment:
I think this could live with the IVC stuff.  I was imagining these utils being largely (entirely) about import/export and not about geometry querying/modification.

Similarly for `merge_layer_surface` but happy to discuss.

Reply:
Separating functionality the way you suggest makes sense to me, but I wonder how this separation of concerns affects the other utility functions.

E.g., with tetrahedral mesh generation (mentioned above) should just the mesh export live within `cubit_utils.py` while the mesh generation lives within `magnet_coils.py`? Similarly, for DAGMC exports, should imprinting/merging and material tagging live within `parastell.py` while `cubit_utils.py` handles only the export?

The legacy and native DAGMC exports handle material tagging and surface faceting differently, so my concern is that this will lead to many rather small functions specific to export methods. That may be fine but I worry it might be overcomplicating things.

Reply:
Also, this routine that skips imprinting depends on STEP files being imported to Cubit, which is called by `parastell.py` and updates component information to include volume IDs for material tagging in Cubit. Because that information belongs to `parastell.py` rather than `invessel_build.py`, I wonder if it might be better to put `merge_layer_surface` and `orient_spline_surfaces` in `parastell.py` rather than `invessel_build.py` as suggested.